### ISSUE TITLE
How to configure logging in blazor webassembly application in 3.2 final release?

### ISSUE THREAD

**Issue by nvmkpk** [dotnet/aspnetcore#22158]
I upgraded from 3.2 preview 2 to the 3.2 final release. I used to have a custom implementation of `ILoggerProvider` and it was working in 3.2 preview 2. After updating to 3.2 final release, it stopped producing any logs in browser console. I also tried removing my custom `ILoggerProvider` but I still can't get logging messages in browser console. I tried few other ways based on how it works for Asp.net Core web application (via appsettings.json etc.) but none of it worked.

This is the only way to debug my application. Blazor WebAssembly debugging in Visual Studio has never worked for me (but that is a different issue).

**Comment by nvmkpk** [comment]
Actually, logging is working if I remove my custom `ILoggerProvider` implementation. But I am unable to configure logging level via appsettings.json file in Asp.net Core's way. I had to add `hostBuilder.Logging.AddConfiguration(hostBuilder.Configuration.GetSection("Logging"));` to make it configurable. Why doesn't WebAssemblyHostBuilder do this by default?

**Comment by mkArtakMSFT** [comment]
@captainsafia can you please answer this? Thanks!

**Comment by captainsafia** [comment]
> Why doesn't WebAssemblyHostBuilder do this by default?

Including this in the default `WebAssemblyHostBuilder` would require that we include the `Logging.Configuration` dependency as part of Blazor which brings in a few other transitive dependencies and increases the overall size of the Blazor app.

We've opted to not include it in the interest of minimizing the standard app size.

We realize that this isn't explicitly documented in the docs so we've got https://github.com/dotnet/AspNetCore.Docs/issues/18444 open to track adding some docs to it.

> I used to have a custom implementation of ILoggerProvider and it was working in 3.2 preview 2. After updating to 3.2 final release, it stopped producing any logs in browser console.

How are you configuring the custom logging provider?
